### PR TITLE
Adafruit Motor HAT driver support

### PR DESCRIPTION
**Pull request by jfinken**
Hello at gobot!  This is a PR to add driver support for the [DC/Stepper Motor HAT](https://www.adafruit.com/products/2348) and [PWM/Servo Motor HAT](https://www.adafruit.com/products/2327) from Adafruit.

A few thoughts:

 * For these HATs, this is a rather direct port of the Adafruit libraries from python.
 * There appears to be over [20 HATs](https://www.adafruit.com/categories/405) for the Pi from Adafruit.  So perhaps the filename for driver should be akin to `motorhat_driver.go`?  I am wide open to suggestions and feedback.
 *  `examples/raspi_adafruit_*.go` are example runner programs.  You probably do not want my little Makefile, I'll `git rm` it.
 * Again I am all too happy to take suggestions and make changes; discuss here, on Slack, IRC etc.  Thanks, and to the maintainers and contributors: great work.

### CODE REVIEW

File: examples/raspi_adafruit.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	"github.com/hybridgroup/gobot/platforms/raspi"
+)
+
+func adafruitStepperMotorRunner(a *i2c.AdafruitMotorHatDriver, motor int) (err error) {
+	log.Printf("Stepper Motor Run Loop...\n")
+	// set the speed state:
+	speed := 30 // rpm
+	a.SetStepperMotorSpeed(motor, speed)
+
+	//fmt.Println("Single coil steps")
```
Comment (deadprogram) [inline examples/raspi_adafruit.go]: Please remove commented out lines :)

**Comment by deadprogram** [comment]
Hi @jfinken thanks very much for working on this. Can you please run `go fmt` on all the files? Also make sure to remove any commented out code. Thank you!

**Comment by jfinken** [comment]
Hi @deadprogram, many thanks for taking a look and getting back to me.  

Good catch regarding `raspi_adafruit.go`.  I previously split that example into `raspi_adafruit_dcmotor.go`, `raspi_adafruit_servo.go` and `raspi_adafruit_stepper.go`, but hadn't yet removed the original, I did that now.

Also, `gofmt` is run on save for files I've modified (as is `govet`, `golint` and `go build`), so I _believe_ the Adafruit code is properly formatted.  That being said I find ~21 files that would be modified on `gofmt` over the entire tree.  Is this what you would like me to do?  I didn't want to step on any toes...

    ~/projects/go/src/github.com/hybridgroup/gobot$ gofmt -l .
    examples/ardrone_face_tracking.go
    examples/bebop.go
    examples/bebop_ps3_video.go
    examples/ble_battery.go
    examples/chip_grove_accelerometer.go
    examples/chip_grove_lcd.go
    examples/chip_mpu6050.go
    examples/edison_button_led.go
    examples/edison_button_led_api.go
    examples/firmata_integration.go
    examples/firmata_mpl115a2.go
    examples/joystick_ps4.go
    examples/minidrone.go
    examples/minidrone_ps3.go
    examples/ollie.go
    examples/opencv_face_detect.go
    examples/opencv_window.go
    examples/sphero_dpad.go
    platforms/bebop/client/constants.go
    platforms/bebop/test_helper.go
    platforms/i2c/mpu6050_driver.go

Thanks!

**Comment by deadprogram** [comment]
Hi @jfinken I must have misread your code. No need to go fmt any files that you did not change.

Thanks!

**Comment by jfinken** [comment]
Hi @deadprogram,  anything I can do to help on this PR (or in general for that matter)?  If this isn't of interest that is totally fine too.

Regarding the failed Travis build, it's a little unclear why it failed.  I do see

    gobottest.go:14: utils_test.go:100: 20 - "int", should equal,  30 - "int"

in the log.  In line 100 of utils_test.go, perhaps it should assert i is 20 not 30.  But as I didn't touch anything there I am reluctant to modify that assertion.  

Thanks,
Josh

**Comment by deadprogram** [comment]
Hi @jfinken that test fail was a spurious one, I think some (all?) of those are fixed in the dev branch.

In any case this is great stuff, thank you very much for the contribution. Merging.

**Comment by jfinken** [comment]
Radical.  Thanks for merging @deadprogram.  I'm continuing to work on projects with gobot as the driver layer so I'll certainly keep tabs on `adafruit_driver`, and gobot in general.  Thanks again.

**Pull request merged**